File: solutions/java/bob/1/src/main/java/Bob.java
import java.util.regex.Pattern;

class Bob {

    String hey(String input) {
        if (input.matches("\s+") || input.isBlank()) {
            return "Fine. Be that way!";
        }
        input = input.replaceAll("[0-9,.'!%^*@#$(*^\s]", "");
        if (input.matches("[A-Z\s]+\\?")) {
            return "Calm down, I know what I'm doing!";
        } else if (input.matches("[A-Z\s]+")) {
            return "Whoa, chill out!";
        } else if (input.endsWith("?")) {
            return "Sure.";
        } else {
            return "Whatever.";
        }
    }
}

File: solutions/typescript/bob/1/bob.ts
export default class Bob {
  private isYelling = (text: string) =>
    text === text.toUpperCase() && /[A-Z]{1,}/.test(text);
  private isAsking = (text: string) => /[?]$/.test(text);
  private isSayingNothing = (text: string) => text === "";

  hey(text: string): string {
    text = text.trim();
    if (this.isYelling(text) && this.isAsking(text)) {
      return "Calm down, I know what I'm doing!";
    } else if (this.isSayingNothing(text)) {
      return "Fine. Be that way!";
    } else if (this.isYelling(text)) {
      return "Whoa, chill out!";
    } else if (this.isAsking(text)) {
      return "Sure.";
    } else {
      return "Whatever.";
    }
  }
}
